File: hackerrank/easy/0057-fair-rations/js/solution.mjs
// solution.mjs

export function fairRations(B) {
  const arr = [...B];
  const oddCount = arr.filter(x => x % 2 !== 0).length;

  if (oddCount % 2 === 1) {
    return "NO";
  }

  let loavesDistributed = 0;

  for (let i = 0; i < arr.length - 1; i++) {
    if (arr[i] % 2 !== 0) {
      arr[i] += 1;
      arr[i + 1] += 1;
      loavesDistributed += 2;
    }
  }

  return String(loavesDistributed);
}

// --- demo run ---
if (import.meta.url === `file://${process.argv[1]}`) {
  console.log(fairRations([2, 3, 4, 5, 6])); // "4"
  console.log(fairRations([1, 2]));          // "NO"
}

File: hackerrank/easy/0057-fair-rations/python/solution.py
## solution.py

from typing import List

def fairRations(B: List[int]) -> str:
    arr = B[:]  # copy
    odd_count = sum(1 for x in arr if x % 2 != 0)

    if odd_count % 2 == 1:
        return "NO"

    loaves_distributed = 0

    for i in range(len(arr) - 1):
        if arr[i] % 2 != 0:
            arr[i] += 1
            arr[i + 1] += 1
            loaves_distributed += 2

    return str(loaves_distributed)


# --- demo run ---
if __name__ == "__main__":
    print(fairRations([2, 3, 4, 5, 6]))  # "4"
    print(fairRations([1, 2]))          # "NO"
